tools/bumper/cnao_repo_commands.go

```diff
@@ -212,6 +215,138 @@ func (cnaoRepoOps *gitCnaoRepo) collectModifiedToTreeList(allowedList []string)
 	return entries, nil
 }
 
+// getBumpBranch creates a new bump branch.
+// Since we don't want to use already open branch
+// that may hold outdated/needs rebasing changes
+// we concat a random string to the end of the branch name
+func (cnaoRepoOps *gitCnaoRepo) getBumpBranch(prTitle string) (*github.Reference, string, error) {
+	branchBaseName := strings.Replace(strings.ToLower(prTitle), " ", "_", -1)
+	branchOpenMaxTries := 10
+	for branchIdx := 0; branchIdx < branchOpenMaxTries; branchIdx++ {
+		branchNameWithRandSuffix := fmt.Sprintf("%s_%s", branchBaseName, rand.String(5))
+		branchRef, _, err := cnaoRepoOps.githubInterface.GetBranchRef(cnaoRepoOps.getCnaoRepoOwnerFromUrl(), cnaoRepoOps.getCnaoRepoNameFromUrl(), "refs/heads/"+branchNameWithRandSuffix)
+		if err != nil {
+			if cnaoRepoOps.branchNotFound(branchRef, err) {
+				return cnaoRepoOps.createNewGithubBranch(branchNameWithRandSuffix)
```

Comment:
We don't retry if this fails ?

Reply:
there are 3 cases here:
1. branch is not yet opened, so create it
2. branch already created. we want to avoid this, so we use random string. but if it does happen we don't want to continue.
3. the case you asked about, where there's a different error. one reason I can think of is the  github is down.. in that case I think we should not retry.. another reason is a conflict, but since we can't  rely on github conflict resolution, I prefer to abort the script.